// @anysphere/proto/dist/generated/sdk/v1/sdk_agent_service_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_agent_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3 } from "@bufbuild/protobuf";
import { AgentMessage, AgentOptions, ModelSelection, RunResult, RunSnapshot, Runtime, SdkAgentInfo, SdkArtifact, SendOptions, UserMessage } from "./sdk_messages_pb.js";
/**
 * @generated from message sdk.v1.CreateAgentRequest
 */
export class CreateAgentRequest extends Message {
    /**
     * @generated from field: sdk.v1.AgentOptions options = 1;
     */
    options;
    /**
     * @generated from field: optional string idempotency_key = 2;
     */
    idempotencyKey;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CreateAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "options", kind: "message", T: AgentOptions },
        { no: 2, name: "idempotency_key", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new CreateAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CreateAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CreateAgentResponse
 */
export class CreateAgentResponse extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.ModelSelection model = 2;
     */
    model;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CreateAgentResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "model", kind: "message", T: ModelSelection },
    ]);
    static fromBinary(bytes, options) {
        return new CreateAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CreateAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CreateAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CreateAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ResumeAgentRequest
 */
export class ResumeAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.AgentOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ResumeAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: AgentOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ResumeAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ResumeAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ResumeAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ResumeAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ResumeAgentResponse
 */
export class ResumeAgentResponse extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.ModelSelection model = 2;
     */
    model;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ResumeAgentResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "model", kind: "message", T: ModelSelection },
    ]);
    static fromBinary(bytes, options) {
        return new ResumeAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ResumeAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ResumeAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ResumeAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ReloadAgentRequest
 */
export class ReloadAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ReloadAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ReloadAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ReloadAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ReloadAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ReloadAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ReloadAgentResponse
 */
export class ReloadAgentResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ReloadAgentResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new ReloadAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ReloadAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ReloadAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ReloadAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloseAgentRequest
 */
export class CloseAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloseAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new CloseAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloseAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloseAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloseAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloseAgentResponse
 */
export class CloseAgentResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloseAgentResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new CloseAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloseAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloseAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloseAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.SendRequest
 */
export class SendRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.UserMessage message = 2;
     */
    message;
    /**
     * @generated from field: sdk.v1.SendOptions options = 3;
     */
    options;
    /**
     * @generated from field: optional string idempotency_key = 4;
     */
    idempotencyKey;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SendRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "message", kind: "message", T: UserMessage },
        { no: 3, name: "options", kind: "message", T: SendOptions },
        { no: 4, name: "idempotency_key", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new SendRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SendRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SendRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SendRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.WaitLiveRunRequest
 */
export class WaitLiveRunRequest extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.WaitLiveRunRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new WaitLiveRunRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new WaitLiveRunRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new WaitLiveRunRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(WaitLiveRunRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.WaitLiveRunResponse
 */
export class WaitLiveRunResponse extends Message {
    /**
     * @generated from field: sdk.v1.RunResult result = 1;
     */
    result;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.WaitLiveRunResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "result", kind: "message", T: RunResult },
    ]);
    static fromBinary(bytes, options) {
        return new WaitLiveRunResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new WaitLiveRunResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new WaitLiveRunResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(WaitLiveRunResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetRunRequest
 */
export class GetRunRequest extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    /**
     * @generated from field: sdk.v1.GetRunOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetRunRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: GetRunOptions },
    ]);
    static fromBinary(bytes, options) {
        return new GetRunRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRunRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRunRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetRunRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetRunResponse
 */
export class GetRunResponse extends Message {
    /**
     * @generated from field: sdk.v1.RunSnapshot run = 1;
     */
    run;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetRunResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run", kind: "message", T: RunSnapshot },
    ]);
    static fromBinary(bytes, options) {
        return new GetRunResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRunResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRunResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetRunResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListRunsRequest
 */
export class ListRunsRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.ListRunsOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListRunsRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: ListRunsOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ListRunsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRunsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRunsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListRunsRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListRunsResponse
 */
export class ListRunsResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.RunSnapshot items = 1;
     */
    items = [];
    /**
     * @generated from field: string next_cursor = 2;
     */
    nextCursor = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListRunsResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "items", kind: "message", T: RunSnapshot, repeated: true },
        { no: 2, name: "next_cursor", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ListRunsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRunsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRunsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListRunsResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetRunConversationRequest
 */
export class GetRunConversationRequest extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetRunConversationRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new GetRunConversationRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRunConversationRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRunConversationRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetRunConversationRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetRunConversationResponse
 */
export class GetRunConversationResponse extends Message {
    /**
     * @generated from field: string conversation_json = 1;
     */
    conversationJson = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetRunConversationResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "conversation_json", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new GetRunConversationResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRunConversationResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRunConversationResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetRunConversationResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ObserveRunRequest
 */
export class ObserveRunRequest extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    /**
     * @generated from field: optional string after_offset = 2;
     */
    afterOffset;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ObserveRunRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "after_offset", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new ObserveRunRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ObserveRunRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ObserveRunRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ObserveRunRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CancelRunRequest
 */
export class CancelRunRequest extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    /**
     * @generated from field: optional string agent_id = 2;
     */
    agentId;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CancelRunRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new CancelRunRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CancelRunRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CancelRunRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CancelRunRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CancelRunResponse
 */
export class CancelRunResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CancelRunResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new CancelRunResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CancelRunResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CancelRunResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CancelRunResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetAgentRequest
 */
export class GetAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.AgentOperationOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: AgentOperationOptions },
    ]);
    static fromBinary(bytes, options) {
        return new GetAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetAgentResponse
 */
export class GetAgentResponse extends Message {
    /**
     * @generated from field: sdk.v1.SdkAgentInfo agent = 1;
     */
    agent;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetAgentResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent", kind: "message", T: SdkAgentInfo },
    ]);
    static fromBinary(bytes, options) {
        return new GetAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListAgentsRequest
 */
export class ListAgentsRequest extends Message {
    /**
     * @generated from field: sdk.v1.ListAgentsOptions options = 1;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListAgentsRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "options", kind: "message", T: ListAgentsOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ListAgentsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListAgentsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListAgentsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListAgentsRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListAgentsResponse
 */
export class ListAgentsResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.SdkAgentInfo items = 1;
     */
    items = [];
    /**
     * @generated from field: string next_cursor = 2;
     */
    nextCursor = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListAgentsResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "items", kind: "message", T: SdkAgentInfo, repeated: true },
        { no: 2, name: "next_cursor", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ListAgentsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListAgentsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListAgentsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListAgentsResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ArchiveAgentRequest
 */
export class ArchiveAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.AgentOperationOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ArchiveAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: AgentOperationOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ArchiveAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ArchiveAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ArchiveAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ArchiveAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ArchiveAgentResponse
 */
export class ArchiveAgentResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ArchiveAgentResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new ArchiveAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ArchiveAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ArchiveAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ArchiveAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.UnarchiveAgentRequest
 */
export class UnarchiveAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.AgentOperationOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.UnarchiveAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: AgentOperationOptions },
    ]);
    static fromBinary(bytes, options) {
        return new UnarchiveAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UnarchiveAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UnarchiveAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(UnarchiveAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.UnarchiveAgentResponse
 */
export class UnarchiveAgentResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.UnarchiveAgentResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new UnarchiveAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UnarchiveAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UnarchiveAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(UnarchiveAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.DeleteAgentRequest
 */
export class DeleteAgentRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.AgentOperationOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.DeleteAgentRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: AgentOperationOptions },
    ]);
    static fromBinary(bytes, options) {
        return new DeleteAgentRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteAgentRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteAgentRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DeleteAgentRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.DeleteAgentResponse
 */
export class DeleteAgentResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.DeleteAgentResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new DeleteAgentResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DeleteAgentResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DeleteAgentResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DeleteAgentResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListAgentMessagesRequest
 */
export class ListAgentMessagesRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.GetAgentMessagesOptions options = 2;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListAgentMessagesRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "options", kind: "message", T: GetAgentMessagesOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ListAgentMessagesRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListAgentMessagesRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListAgentMessagesRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListAgentMessagesRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListAgentMessagesResponse
 */
export class ListAgentMessagesResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.AgentMessage messages = 1;
     */
    messages = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListAgentMessagesResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "messages", kind: "message", T: AgentMessage, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ListAgentMessagesResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListAgentMessagesResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListAgentMessagesResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListAgentMessagesResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListArtifactsRequest
 */
export class ListArtifactsRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListArtifactsRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ListArtifactsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListArtifactsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListArtifactsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListArtifactsRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListArtifactsResponse
 */
export class ListArtifactsResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.SdkArtifact artifacts = 1;
     */
    artifacts = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListArtifactsResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "artifacts", kind: "message", T: SdkArtifact, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ListArtifactsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListArtifactsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListArtifactsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListArtifactsResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.DownloadArtifactRequest
 */
export class DownloadArtifactRequest extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: string path = 2;
     */
    path = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.DownloadArtifactRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "path", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new DownloadArtifactRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DownloadArtifactRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DownloadArtifactRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DownloadArtifactRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListAgentsOptions
 */
export class ListAgentsOptions extends Message {
    /**
     * @generated from field: uint32 limit = 1;
     */
    limit = 0;
    /**
     * @generated from field: string cursor = 2;
     */
    cursor = "";
    /**
     * @generated from field: sdk.v1.Runtime runtime = 3;
     */
    runtime = Runtime.UNSPECIFIED;
    /**
     * @generated from field: string cwd = 4;
     */
    cwd = "";
    /**
     * @generated from field: string pr_url = 5;
     */
    prUrl = "";
    /**
     * @generated from field: optional bool include_archived = 6;
     */
    includeArchived;
    /**
     * @generated from field: string api_key = 7;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListAgentsOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "limit", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
        { no: 2, name: "cursor", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "runtime", kind: "enum", T: proto3.getEnumType(Runtime) },
        { no: 4, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "pr_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 6, name: "include_archived", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 7, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ListAgentsOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListAgentsOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListAgentsOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListAgentsOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListRunsOptions
 */
export class ListRunsOptions extends Message {
    /**
     * @generated from field: uint32 limit = 1;
     */
    limit = 0;
    /**
     * @generated from field: string cursor = 2;
     */
    cursor = "";
    /**
     * @generated from field: sdk.v1.Runtime runtime = 3;
     */
    runtime = Runtime.UNSPECIFIED;
    /**
     * @generated from field: string cwd = 4;
     */
    cwd = "";
    /**
     * @generated from field: string api_key = 5;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListRunsOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "limit", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
        { no: 2, name: "cursor", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "runtime", kind: "enum", T: proto3.getEnumType(Runtime) },
        { no: 4, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ListRunsOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRunsOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRunsOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListRunsOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetRunOptions
 */
export class GetRunOptions extends Message {
    /**
     * @generated from field: sdk.v1.Runtime runtime = 1;
     */
    runtime = Runtime.UNSPECIFIED;
    /**
     * @generated from field: string cwd = 2;
     */
    cwd = "";
    /**
     * @generated from field: string agent_id = 3;
     */
    agentId = "";
    /**
     * @generated from field: string api_key = 4;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetRunOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "runtime", kind: "enum", T: proto3.getEnumType(Runtime) },
        { no: 2, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new GetRunOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetRunOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetRunOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetRunOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.AgentOperationOptions
 */
export class AgentOperationOptions extends Message {
    /**
     * @generated from field: string cwd = 1;
     */
    cwd = "";
    /**
     * @generated from field: string api_key = 2;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.AgentOperationOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new AgentOperationOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AgentOperationOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AgentOperationOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AgentOperationOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetAgentMessagesOptions
 */
export class GetAgentMessagesOptions extends Message {
    /**
     * @generated from field: uint32 limit = 1;
     */
    limit = 0;
    /**
     * @generated from field: uint32 offset = 2;
     */
    offset = 0;
    /**
     * @generated from field: sdk.v1.Runtime runtime = 3;
     */
    runtime = Runtime.UNSPECIFIED;
    /**
     * @generated from field: string cwd = 4;
     */
    cwd = "";
    /**
     * @generated from field: string api_key = 5;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetAgentMessagesOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "limit", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
        { no: 2, name: "offset", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
        { no: 3, name: "runtime", kind: "enum", T: proto3.getEnumType(Runtime) },
        { no: 4, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new GetAgentMessagesOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetAgentMessagesOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetAgentMessagesOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetAgentMessagesOptions, a, b);
    }
}
