// @anysphere/proto/dist/generated/sdk/v1/sdk_custom_tool_callback_service_connect.js
// @generated by protoc-gen-connect-es v1.6.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_custom_tool_callback_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { CallCustomToolRequest, CallCustomToolResponse } from "./sdk_custom_tool_callback_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
/**
 * Custom-tool callbacks from the Node bridge to a Python (or other) host process.
 *
 * The bridge forwards in-process SDK custom-tool executions to the host over
 * `CallCustomTool`, mirroring `@cursor/sdk` `local.customTools` execute
 * callbacks. Tool metadata crosses CreateAgent/Send options; only execution
 * round-trips to the host.
 *
 * @generated from service sdk.v1.SdkCustomToolCallbackService
 */
export const SdkCustomToolCallbackService = {
    typeName: "sdk.v1.SdkCustomToolCallbackService",
    methods: {
        /**
         * @generated from rpc sdk.v1.SdkCustomToolCallbackService.CallCustomTool
         */
        callCustomTool: {
            name: "CallCustomTool",
            I: CallCustomToolRequest,
            O: CallCustomToolResponse,
            kind: MethodKind.Unary,
        },
    }
};
