// @anysphere/proto/dist/generated/sdk/v1/sdk_cursor_service_connect.js
// @generated by protoc-gen-connect-es v1.6.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_cursor_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { ListModelsRequest, ListModelsResponse, ListRepositoriesRequest, ListRepositoriesResponse, MeRequest, MeResponse } from "./sdk_cursor_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
/**
 * Cloud-only Cursor account and catalog APIs.
 *
 * @generated from service sdk.v1.SdkCursorService
 */
export const SdkCursorService = {
    typeName: "sdk.v1.SdkCursorService",
    methods: {
        /**
         * @generated from rpc sdk.v1.SdkCursorService.Me
         */
        me: {
            name: "Me",
            I: MeRequest,
            O: MeResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkCursorService.ListModels
         */
        listModels: {
            name: "ListModels",
            I: ListModelsRequest,
            O: ListModelsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkCursorService.ListRepositories
         */
        listRepositories: {
            name: "ListRepositories",
            I: ListRepositoriesRequest,
            O: ListRepositoriesResponse,
            kind: MethodKind.Unary,
        },
    }
};
