// @anysphere/proto/dist/generated/sdk/v1/sdk_messages_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_messages.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3, protoInt64, Struct, Timestamp } from "@bufbuild/protobuf";
export { Runtime };
/**
 * Runtime selector for SDK APIs that can route to local or cloud agents.
 *
 * @generated from enum sdk.v1.Runtime
 */
var Runtime;
(function (Runtime) {
    /**
     * @generated from enum value: RUNTIME_UNSPECIFIED = 0;
     */
    Runtime[Runtime["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: RUNTIME_AUTO = 1;
     */
    Runtime[Runtime["AUTO"] = 1] = "AUTO";
    /**
     * @generated from enum value: RUNTIME_LOCAL = 2;
     */
    Runtime[Runtime["LOCAL"] = 2] = "LOCAL";
    /**
     * @generated from enum value: RUNTIME_CLOUD = 3;
     */
    Runtime[Runtime["CLOUD"] = 3] = "CLOUD";
})(Runtime || (Runtime = {}));
// Retrieve enum metadata with: proto3.getEnumType(Runtime)
proto3.util.setEnumType(Runtime, "sdk.v1.Runtime", [
    { no: 0, name: "RUNTIME_UNSPECIFIED" },
    { no: 1, name: "RUNTIME_AUTO" },
    { no: 2, name: "RUNTIME_LOCAL" },
    { no: 3, name: "RUNTIME_CLOUD" },
]);
export { RunLifecycleStatus };
/**
 * Lifecycle status used for run snapshots and stream status messages.
 *
 * @generated from enum sdk.v1.RunLifecycleStatus
 */
var RunLifecycleStatus;
(function (RunLifecycleStatus) {
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_UNSPECIFIED = 0;
     */
    RunLifecycleStatus[RunLifecycleStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_CREATING = 1;
     */
    RunLifecycleStatus[RunLifecycleStatus["CREATING"] = 1] = "CREATING";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_RUNNING = 2;
     */
    RunLifecycleStatus[RunLifecycleStatus["RUNNING"] = 2] = "RUNNING";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_FINISHED = 3;
     */
    RunLifecycleStatus[RunLifecycleStatus["FINISHED"] = 3] = "FINISHED";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_ERROR = 4;
     */
    RunLifecycleStatus[RunLifecycleStatus["ERROR"] = 4] = "ERROR";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_CANCELLED = 5;
     */
    RunLifecycleStatus[RunLifecycleStatus["CANCELLED"] = 5] = "CANCELLED";
    /**
     * @generated from enum value: RUN_LIFECYCLE_STATUS_EXPIRED = 6;
     */
    RunLifecycleStatus[RunLifecycleStatus["EXPIRED"] = 6] = "EXPIRED";
})(RunLifecycleStatus || (RunLifecycleStatus = {}));
// Retrieve enum metadata with: proto3.getEnumType(RunLifecycleStatus)
proto3.util.setEnumType(RunLifecycleStatus, "sdk.v1.RunLifecycleStatus", [
    { no: 0, name: "RUN_LIFECYCLE_STATUS_UNSPECIFIED" },
    { no: 1, name: "RUN_LIFECYCLE_STATUS_CREATING" },
    { no: 2, name: "RUN_LIFECYCLE_STATUS_RUNNING" },
    { no: 3, name: "RUN_LIFECYCLE_STATUS_FINISHED" },
    { no: 4, name: "RUN_LIFECYCLE_STATUS_ERROR" },
    { no: 5, name: "RUN_LIFECYCLE_STATUS_CANCELLED" },
    { no: 6, name: "RUN_LIFECYCLE_STATUS_EXPIRED" },
]);
export { AgentInfoStatus };
/**
 * Agent lifecycle status returned by list/get agent calls.
 *
 * @generated from enum sdk.v1.AgentInfoStatus
 */
var AgentInfoStatus;
(function (AgentInfoStatus) {
    /**
     * @generated from enum value: AGENT_INFO_STATUS_UNSPECIFIED = 0;
     */
    AgentInfoStatus[AgentInfoStatus["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: AGENT_INFO_STATUS_RUNNING = 1;
     */
    AgentInfoStatus[AgentInfoStatus["RUNNING"] = 1] = "RUNNING";
    /**
     * @generated from enum value: AGENT_INFO_STATUS_FINISHED = 2;
     */
    AgentInfoStatus[AgentInfoStatus["FINISHED"] = 2] = "FINISHED";
    /**
     * @generated from enum value: AGENT_INFO_STATUS_ERROR = 3;
     */
    AgentInfoStatus[AgentInfoStatus["ERROR"] = 3] = "ERROR";
})(AgentInfoStatus || (AgentInfoStatus = {}));
// Retrieve enum metadata with: proto3.getEnumType(AgentInfoStatus)
proto3.util.setEnumType(AgentInfoStatus, "sdk.v1.AgentInfoStatus", [
    { no: 0, name: "AGENT_INFO_STATUS_UNSPECIFIED" },
    { no: 1, name: "AGENT_INFO_STATUS_RUNNING" },
    { no: 2, name: "AGENT_INFO_STATUS_FINISHED" },
    { no: 3, name: "AGENT_INFO_STATUS_ERROR" },
]);
export { SettingSource };
/**
 * Cursor setting sources accepted by local agent options.
 *
 * @generated from enum sdk.v1.SettingSource
 */
var SettingSource;
(function (SettingSource) {
    /**
     * @generated from enum value: SETTING_SOURCE_UNSPECIFIED = 0;
     */
    SettingSource[SettingSource["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: SETTING_SOURCE_PROJECT = 1;
     */
    SettingSource[SettingSource["PROJECT"] = 1] = "PROJECT";
    /**
     * @generated from enum value: SETTING_SOURCE_USER = 2;
     */
    SettingSource[SettingSource["USER"] = 2] = "USER";
    /**
     * @generated from enum value: SETTING_SOURCE_TEAM = 3;
     */
    SettingSource[SettingSource["TEAM"] = 3] = "TEAM";
    /**
     * @generated from enum value: SETTING_SOURCE_MDM = 4;
     */
    SettingSource[SettingSource["MDM"] = 4] = "MDM";
    /**
     * @generated from enum value: SETTING_SOURCE_PLUGINS = 5;
     */
    SettingSource[SettingSource["PLUGINS"] = 5] = "PLUGINS";
    /**
     * @generated from enum value: SETTING_SOURCE_ALL = 6;
     */
    SettingSource[SettingSource["ALL"] = 6] = "ALL";
})(SettingSource || (SettingSource = {}));
// Retrieve enum metadata with: proto3.getEnumType(SettingSource)
proto3.util.setEnumType(SettingSource, "sdk.v1.SettingSource", [
    { no: 0, name: "SETTING_SOURCE_UNSPECIFIED" },
    { no: 1, name: "SETTING_SOURCE_PROJECT" },
    { no: 2, name: "SETTING_SOURCE_USER" },
    { no: 3, name: "SETTING_SOURCE_TEAM" },
    { no: 4, name: "SETTING_SOURCE_MDM" },
    { no: 5, name: "SETTING_SOURCE_PLUGINS" },
    { no: 6, name: "SETTING_SOURCE_ALL" },
]);
export { CloudEnvironmentType };
/**
 * Cloud execution environment type.
 *
 * @generated from enum sdk.v1.CloudEnvironmentType
 */
var CloudEnvironmentType;
(function (CloudEnvironmentType) {
    /**
     * @generated from enum value: CLOUD_ENVIRONMENT_TYPE_UNSPECIFIED = 0;
     */
    CloudEnvironmentType[CloudEnvironmentType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: CLOUD_ENVIRONMENT_TYPE_CLOUD = 1;
     */
    CloudEnvironmentType[CloudEnvironmentType["CLOUD"] = 1] = "CLOUD";
    /**
     * @generated from enum value: CLOUD_ENVIRONMENT_TYPE_POOL = 2;
     */
    CloudEnvironmentType[CloudEnvironmentType["POOL"] = 2] = "POOL";
    /**
     * @generated from enum value: CLOUD_ENVIRONMENT_TYPE_MACHINE = 3;
     */
    CloudEnvironmentType[CloudEnvironmentType["MACHINE"] = 3] = "MACHINE";
})(CloudEnvironmentType || (CloudEnvironmentType = {}));
// Retrieve enum metadata with: proto3.getEnumType(CloudEnvironmentType)
proto3.util.setEnumType(CloudEnvironmentType, "sdk.v1.CloudEnvironmentType", [
    { no: 0, name: "CLOUD_ENVIRONMENT_TYPE_UNSPECIFIED" },
    { no: 1, name: "CLOUD_ENVIRONMENT_TYPE_CLOUD" },
    { no: 2, name: "CLOUD_ENVIRONMENT_TYPE_POOL" },
    { no: 3, name: "CLOUD_ENVIRONMENT_TYPE_MACHINE" },
]);
export { HttpMcpTransportType };
/**
 * MCP transport used for HTTP-style MCP servers.
 *
 * @generated from enum sdk.v1.HttpMcpTransportType
 */
var HttpMcpTransportType;
(function (HttpMcpTransportType) {
    /**
     * @generated from enum value: HTTP_MCP_TRANSPORT_TYPE_UNSPECIFIED = 0;
     */
    HttpMcpTransportType[HttpMcpTransportType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: HTTP_MCP_TRANSPORT_TYPE_HTTP = 1;
     */
    HttpMcpTransportType[HttpMcpTransportType["HTTP"] = 1] = "HTTP";
    /**
     * @generated from enum value: HTTP_MCP_TRANSPORT_TYPE_SSE = 2;
     */
    HttpMcpTransportType[HttpMcpTransportType["SSE"] = 2] = "SSE";
})(HttpMcpTransportType || (HttpMcpTransportType = {}));
// Retrieve enum metadata with: proto3.getEnumType(HttpMcpTransportType)
proto3.util.setEnumType(HttpMcpTransportType, "sdk.v1.HttpMcpTransportType", [
    { no: 0, name: "HTTP_MCP_TRANSPORT_TYPE_UNSPECIFIED" },
    { no: 1, name: "HTTP_MCP_TRANSPORT_TYPE_HTTP" },
    { no: 2, name: "HTTP_MCP_TRANSPORT_TYPE_SSE" },
]);
export { AgentModeOption };
/**
 * SDK conversation mode for agent creation and per-send overrides.
 *
 * @generated from enum sdk.v1.AgentModeOption
 */
var AgentModeOption;
(function (AgentModeOption) {
    /**
     * @generated from enum value: AGENT_MODE_OPTION_UNSPECIFIED = 0;
     */
    AgentModeOption[AgentModeOption["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: AGENT_MODE_OPTION_AGENT = 1;
     */
    AgentModeOption[AgentModeOption["AGENT"] = 1] = "AGENT";
    /**
     * @generated from enum value: AGENT_MODE_OPTION_PLAN = 2;
     */
    AgentModeOption[AgentModeOption["PLAN"] = 2] = "PLAN";
})(AgentModeOption || (AgentModeOption = {}));
// Retrieve enum metadata with: proto3.getEnumType(AgentModeOption)
proto3.util.setEnumType(AgentModeOption, "sdk.v1.AgentModeOption", [
    { no: 0, name: "AGENT_MODE_OPTION_UNSPECIFIED" },
    { no: 1, name: "AGENT_MODE_OPTION_AGENT" },
    { no: 2, name: "AGENT_MODE_OPTION_PLAN" },
]);
/**
 * @generated from message sdk.v1.ModelSelection
 */
export class ModelSelection extends Message {
    /**
     * @generated from field: string id = 1;
     */
    id = "";
    /**
     * @generated from field: repeated sdk.v1.ModelParameterValue params = 2;
     */
    params = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ModelSelection";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "params", kind: "message", T: ModelParameterValue, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ModelSelection().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ModelSelection().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ModelSelection().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ModelSelection, a, b);
    }
}
/**
 * @generated from message sdk.v1.ModelParameterValue
 */
export class ModelParameterValue extends Message {
    /**
     * @generated from field: string id = 1;
     */
    id = "";
    /**
     * @generated from field: string value = 2;
     */
    value = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ModelParameterValue";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "value", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ModelParameterValue().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ModelParameterValue().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ModelParameterValue().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ModelParameterValue, a, b);
    }
}
/**
 * @generated from message sdk.v1.ModelParameterDefinition
 */
export class ModelParameterDefinition extends Message {
    /**
     * @generated from field: string id = 1;
     */
    id = "";
    /**
     * @generated from field: string display_name = 2;
     */
    displayName = "";
    /**
     * @generated from field: repeated sdk.v1.ModelParameterDefinitionValue values = 3;
     */
    values = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ModelParameterDefinition";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "display_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "values", kind: "message", T: ModelParameterDefinitionValue, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ModelParameterDefinition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ModelParameterDefinition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ModelParameterDefinition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ModelParameterDefinition, a, b);
    }
}
/**
 * @generated from message sdk.v1.ModelParameterDefinitionValue
 */
export class ModelParameterDefinitionValue extends Message {
    /**
     * @generated from field: string value = 1;
     */
    value = "";
    /**
     * @generated from field: string display_name = 2;
     */
    displayName = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ModelParameterDefinitionValue";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "value", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "display_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new ModelParameterDefinitionValue().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ModelParameterDefinitionValue().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ModelParameterDefinitionValue().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ModelParameterDefinitionValue, a, b);
    }
}
/**
 * @generated from message sdk.v1.ModelVariant
 */
export class ModelVariant extends Message {
    /**
     * @generated from field: repeated sdk.v1.ModelParameterValue params = 1;
     */
    params = [];
    /**
     * @generated from field: string display_name = 2;
     */
    displayName = "";
    /**
     * @generated from field: string description = 3;
     */
    description = "";
    /**
     * @generated from field: bool is_default = 4;
     */
    isDefault = false;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ModelVariant";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "params", kind: "message", T: ModelParameterValue, repeated: true },
        { no: 2, name: "display_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "is_default", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
    ]);
    static fromBinary(bytes, options) {
        return new ModelVariant().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ModelVariant().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ModelVariant().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ModelVariant, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkModel
 */
export class SdkModel extends Message {
    /**
     * @generated from field: string id = 1;
     */
    id = "";
    /**
     * @generated from field: string display_name = 2;
     */
    displayName = "";
    /**
     * @generated from field: string description = 3;
     */
    description = "";
    /**
     * @generated from field: repeated sdk.v1.ModelParameterDefinition parameters = 4;
     */
    parameters = [];
    /**
     * @generated from field: repeated sdk.v1.ModelVariant variants = 5;
     */
    variants = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkModel";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "display_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "parameters", kind: "message", T: ModelParameterDefinition, repeated: true },
        { no: 5, name: "variants", kind: "message", T: ModelVariant, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new SdkModel().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkModel().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkModel().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkModel, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkUser
 */
export class SdkUser extends Message {
    /**
     * @generated from field: string api_key_name = 1;
     */
    apiKeyName = "";
    /**
     * @generated from field: uint64 user_id = 2;
     */
    userId = protoInt64.zero;
    /**
     * @generated from field: string user_email = 3;
     */
    userEmail = "";
    /**
     * @generated from field: string user_first_name = 4;
     */
    userFirstName = "";
    /**
     * @generated from field: string user_last_name = 5;
     */
    userLastName = "";
    /**
     * @generated from field: string created_at = 6;
     */
    createdAt = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkUser";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "api_key_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "user_id", kind: "scalar", T: 4 /* ScalarType.UINT64 */ },
        { no: 3, name: "user_email", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "user_first_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "user_last_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 6, name: "created_at", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkUser().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkUser().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkUser().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkUser, a, b);
    }
}
/**
 * Current @cursor/sdk repository shape. Rich metadata can be added later.
 *
 * @generated from message sdk.v1.SdkRepository
 */
export class SdkRepository extends Message {
    /**
     * @generated from field: string url = 1;
     */
    url = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkRepository";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkRepository().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkRepository().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkRepository().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkRepository, a, b);
    }
}
/**
 * @generated from message sdk.v1.UserMessage
 */
export class UserMessage extends Message {
    /**
     * @generated from field: string text = 1;
     */
    text = "";
    /**
     * @generated from field: repeated sdk.v1.SdkImage images = 2;
     */
    images = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.UserMessage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "text", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "images", kind: "message", T: SdkImage, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new UserMessage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new UserMessage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new UserMessage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(UserMessage, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkImage
 */
export class SdkImage extends Message {
    /**
     * @generated from oneof sdk.v1.SdkImage.source
     */
    source = { case: undefined };
    /**
     * @generated from field: sdk.v1.SdkImageDimension dimension = 3;
     */
    dimension;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkImage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "url", kind: "message", T: SdkImageUrl, oneof: "source" },
        { no: 2, name: "data", kind: "message", T: SdkImageData, oneof: "source" },
        { no: 3, name: "dimension", kind: "message", T: SdkImageDimension },
    ]);
    static fromBinary(bytes, options) {
        return new SdkImage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkImage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkImage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkImage, a, b);
    }
}
/**
 * URL images are accepted only by cloud-routed agents in v1.
 *
 * @generated from message sdk.v1.SdkImageUrl
 */
export class SdkImageUrl extends Message {
    /**
     * @generated from field: string url = 1;
     */
    url = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkImageUrl";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkImageUrl().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkImageUrl().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkImageUrl().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkImageUrl, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkImageData
 */
export class SdkImageData extends Message {
    /**
     * @generated from field: string data = 1;
     */
    data = "";
    /**
     * @generated from field: string mime_type = 2;
     */
    mimeType = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkImageData";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "data", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "mime_type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkImageData().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkImageData().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkImageData().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkImageData, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkImageDimension
 */
export class SdkImageDimension extends Message {
    /**
     * @generated from field: uint32 width = 1;
     */
    width = 0;
    /**
     * @generated from field: uint32 height = 2;
     */
    height = 0;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkImageDimension";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "width", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
        { no: 2, name: "height", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkImageDimension().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkImageDimension().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkImageDimension().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkImageDimension, a, b);
    }
}
/**
 * @generated from message sdk.v1.AgentOptions
 */
export class AgentOptions extends Message {
    /**
     * @generated from field: sdk.v1.ModelSelection model = 1;
     */
    model;
    /**
     * @generated from field: string api_key = 2;
     */
    apiKey = "";
    /**
     * @generated from field: string name = 3;
     */
    name = "";
    /**
     * @generated from field: sdk.v1.LocalAgentOptions local = 4;
     */
    local;
    /**
     * @generated from field: sdk.v1.CloudAgentOptions cloud = 5;
     */
    cloud;
    /**
     * @generated from field: map<string, sdk.v1.McpServerConfig> mcp_servers = 6;
     */
    mcpServers = {};
    /**
     * @generated from field: map<string, sdk.v1.AgentDefinition> agents = 7;
     */
    agents = {};
    /**
     * @generated from field: string agent_id = 8;
     */
    agentId = "";
    /**
     * Initial conversation mode for the agent.
     *
     * @generated from field: sdk.v1.AgentModeOption mode = 9;
     */
    mode = AgentModeOption.UNSPECIFIED;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.AgentOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "model", kind: "message", T: ModelSelection },
        { no: 2, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "local", kind: "message", T: LocalAgentOptions },
        { no: 5, name: "cloud", kind: "message", T: CloudAgentOptions },
        { no: 6, name: "mcp_servers", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: McpServerConfig } },
        { no: 7, name: "agents", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: AgentDefinition } },
        { no: 8, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 9, name: "mode", kind: "enum", T: proto3.getEnumType(AgentModeOption) },
    ]);
    static fromBinary(bytes, options) {
        return new AgentOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AgentOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AgentOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AgentOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.SendOptions
 */
export class SendOptions extends Message {
    /**
     * @generated from field: sdk.v1.ModelSelection model = 1;
     */
    model;
    /**
     * @generated from field: map<string, sdk.v1.McpServerConfig> mcp_servers = 2;
     */
    mcpServers = {};
    /**
     * @generated from field: sdk.v1.LocalSendOptions local = 3;
     */
    local;
    /**
     * When true, the bridge forwards raw InteractionUpdate callbacks from
     * @cursor/sdk as RunStreamMessage.interaction_update events.
     *
     * @generated from field: bool enable_deltas = 4;
     */
    enableDeltas = false;
    /**
     * When true, the bridge forwards completed ConversationStep callbacks from
     * @cursor/sdk as RunStreamMessage.step events.
     *
     * @generated from field: bool enable_steps = 5;
     */
    enableSteps = false;
    /**
     * Per-send conversation mode override.
     *
     * @generated from field: sdk.v1.AgentModeOption mode = 6;
     */
    mode = AgentModeOption.UNSPECIFIED;
    /**
     * Cloud-only per-send options.
     *
     * @generated from field: sdk.v1.CloudSendOptions cloud = 7;
     */
    cloud;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SendOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "model", kind: "message", T: ModelSelection },
        { no: 2, name: "mcp_servers", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: McpServerConfig } },
        { no: 3, name: "local", kind: "message", T: LocalSendOptions },
        { no: 4, name: "enable_deltas", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
        { no: 5, name: "enable_steps", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
        { no: 6, name: "mode", kind: "enum", T: proto3.getEnumType(AgentModeOption) },
        { no: 7, name: "cloud", kind: "message", T: CloudSendOptions },
    ]);
    static fromBinary(bytes, options) {
        return new SendOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SendOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SendOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SendOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.LocalSendOptions
 */
export class LocalSendOptions extends Message {
    /**
     * Local-only force send; cloud-routed agents reject this field.
     *
     * @generated from field: optional bool force = 1;
     */
    force;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.LocalSendOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "force", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new LocalSendOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LocalSendOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LocalSendOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(LocalSendOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloudSendOptions
 */
export class CloudSendOptions extends Message {
    /**
     * Run-scoped env vars. Applied only while this send's run executes.
     *
     * @generated from field: map<string, string> env_vars = 1;
     */
    envVars = {};
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloudSendOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "env_vars", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "scalar", T: 9 /* ScalarType.STRING */ } },
    ]);
    static fromBinary(bytes, options) {
        return new CloudSendOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloudSendOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloudSendOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloudSendOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.LocalAgentStoreConfig
 */
export class LocalAgentStoreConfig extends Message {
    /**
     * Store topology: "sqlite" (bridge-owned SQLite; also the default when unset),
     * "jsonl" (bridge-owned JSONL files), or "custom" (fully host-owned via the
     * SdkStoreCallbackService).
     *
     * @generated from field: string type = 1;
     */
    type = "";
    /**
     * Required for "jsonl".
     *
     * @generated from field: string root_dir = 2;
     */
    rootDir = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.LocalAgentStoreConfig";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "root_dir", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new LocalAgentStoreConfig().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LocalAgentStoreConfig().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LocalAgentStoreConfig().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(LocalAgentStoreConfig, a, b);
    }
}
/**
 * @generated from message sdk.v1.CustomToolDefinition
 */
export class CustomToolDefinition extends Message {
    /**
     * @generated from field: optional string description = 1;
     */
    description;
    /**
     * @generated from field: google.protobuf.Struct input_schema = 2;
     */
    inputSchema;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CustomToolDefinition";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 2, name: "input_schema", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new CustomToolDefinition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CustomToolDefinition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CustomToolDefinition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CustomToolDefinition, a, b);
    }
}
/**
 * @generated from message sdk.v1.LocalAgentOptions
 */
export class LocalAgentOptions extends Message {
    /**
     * @generated from field: repeated string cwd = 1;
     */
    cwd = [];
    /**
     * @generated from field: repeated sdk.v1.SettingSource setting_sources = 2;
     */
    settingSources = [];
    /**
     * @generated from field: sdk.v1.SandboxOptions sandbox_options = 3;
     */
    sandboxOptions;
    /**
     * @generated from field: sdk.v1.LocalAgentStoreConfig store = 4;
     */
    store;
    /**
     * When true, local runs use Smart Auto Review (classifier-backed Auto) for tool calls.
     *
     * @generated from field: optional bool auto_review = 5;
     */
    autoReview;
    /**
     * Agent-level custom tools (metadata only). Execution is host-owned via
     * SdkCustomToolCallbackService when using the Python SDK bridge.
     *
     * @generated from field: map<string, sdk.v1.CustomToolDefinition> custom_tools = 6;
     */
    customTools = {};
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.LocalAgentOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
        { no: 2, name: "setting_sources", kind: "enum", T: proto3.getEnumType(SettingSource), repeated: true },
        { no: 3, name: "sandbox_options", kind: "message", T: SandboxOptions },
        { no: 4, name: "store", kind: "message", T: LocalAgentStoreConfig },
        { no: 5, name: "auto_review", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 6, name: "custom_tools", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "message", T: CustomToolDefinition } },
    ]);
    static fromBinary(bytes, options) {
        return new LocalAgentOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LocalAgentOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LocalAgentOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(LocalAgentOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.SandboxOptions
 */
export class SandboxOptions extends Message {
    /**
     * @generated from field: optional bool enabled = 1;
     */
    enabled;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SandboxOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "enabled", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new SandboxOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SandboxOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SandboxOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SandboxOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloudAgentOptions
 */
export class CloudAgentOptions extends Message {
    /**
     * @generated from field: sdk.v1.CloudEnvironment env = 1;
     */
    env;
    /**
     * @generated from field: repeated sdk.v1.CloudRepository repos = 2;
     */
    repos = [];
    /**
     * @generated from field: optional bool work_on_current_branch = 3;
     */
    workOnCurrentBranch;
    /**
     * @generated from field: optional bool auto_create_pr = 4;
     */
    autoCreatePr;
    /**
     * @generated from field: optional bool skip_reviewer_request = 5;
     */
    skipReviewerRequest;
    /**
     * @generated from field: map<string, string> env_vars = 6;
     */
    envVars = {};
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloudAgentOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "env", kind: "message", T: CloudEnvironment },
        { no: 2, name: "repos", kind: "message", T: CloudRepository, repeated: true },
        { no: 3, name: "work_on_current_branch", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 4, name: "auto_create_pr", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 5, name: "skip_reviewer_request", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 6, name: "env_vars", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "scalar", T: 9 /* ScalarType.STRING */ } },
    ]);
    static fromBinary(bytes, options) {
        return new CloudAgentOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloudAgentOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloudAgentOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloudAgentOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloudEnvironment
 */
export class CloudEnvironment extends Message {
    /**
     * @generated from field: sdk.v1.CloudEnvironmentType type = 1;
     */
    type = CloudEnvironmentType.UNSPECIFIED;
    /**
     * @generated from field: string name = 2;
     */
    name = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloudEnvironment";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "enum", T: proto3.getEnumType(CloudEnvironmentType) },
        { no: 2, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new CloudEnvironment().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloudEnvironment().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloudEnvironment().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloudEnvironment, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloudRepository
 */
export class CloudRepository extends Message {
    /**
     * @generated from field: string url = 1;
     */
    url = "";
    /**
     * @generated from field: string starting_ref = 2;
     */
    startingRef = "";
    /**
     * @generated from field: string pr_url = 3;
     */
    prUrl = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloudRepository";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "starting_ref", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "pr_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new CloudRepository().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloudRepository().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloudRepository().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloudRepository, a, b);
    }
}
/**
 * @generated from message sdk.v1.McpServerConfig
 */
export class McpServerConfig extends Message {
    /**
     * @generated from oneof sdk.v1.McpServerConfig.config
     */
    config = { case: undefined };
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.McpServerConfig";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "stdio", kind: "message", T: StdioMcpServerConfig, oneof: "config" },
        { no: 2, name: "http", kind: "message", T: HttpMcpServerConfig, oneof: "config" },
    ]);
    static fromBinary(bytes, options) {
        return new McpServerConfig().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new McpServerConfig().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new McpServerConfig().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(McpServerConfig, a, b);
    }
}
/**
 * @generated from message sdk.v1.StdioMcpServerConfig
 */
export class StdioMcpServerConfig extends Message {
    /**
     * @generated from field: string command = 1;
     */
    command = "";
    /**
     * @generated from field: repeated string args = 2;
     */
    args = [];
    /**
     * @generated from field: map<string, string> env = 3;
     */
    env = {};
    /**
     * @generated from field: string cwd = 4;
     */
    cwd = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.StdioMcpServerConfig";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "command", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "args", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
        { no: 3, name: "env", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "scalar", T: 9 /* ScalarType.STRING */ } },
        { no: 4, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new StdioMcpServerConfig().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new StdioMcpServerConfig().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new StdioMcpServerConfig().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(StdioMcpServerConfig, a, b);
    }
}
/**
 * @generated from message sdk.v1.HttpMcpServerConfig
 */
export class HttpMcpServerConfig extends Message {
    /**
     * @generated from field: sdk.v1.HttpMcpTransportType type = 1;
     */
    type = HttpMcpTransportType.UNSPECIFIED;
    /**
     * @generated from field: string url = 2;
     */
    url = "";
    /**
     * @generated from field: map<string, string> headers = 3;
     */
    headers = {};
    /**
     * @generated from field: sdk.v1.McpAuthConfig auth = 4;
     */
    auth;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.HttpMcpServerConfig";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "enum", T: proto3.getEnumType(HttpMcpTransportType) },
        { no: 2, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "headers", kind: "map", K: 9 /* ScalarType.STRING */, V: { kind: "scalar", T: 9 /* ScalarType.STRING */ } },
        { no: 4, name: "auth", kind: "message", T: McpAuthConfig },
    ]);
    static fromBinary(bytes, options) {
        return new HttpMcpServerConfig().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new HttpMcpServerConfig().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new HttpMcpServerConfig().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(HttpMcpServerConfig, a, b);
    }
}
/**
 * @generated from message sdk.v1.McpAuthConfig
 */
export class McpAuthConfig extends Message {
    /**
     * @generated from field: string client_id = 1;
     */
    clientId = "";
    /**
     * @generated from field: string client_secret = 2;
     */
    clientSecret = "";
    /**
     * @generated from field: repeated string scopes = 3;
     */
    scopes = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.McpAuthConfig";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "client_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "client_secret", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "scopes", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new McpAuthConfig().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new McpAuthConfig().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new McpAuthConfig().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(McpAuthConfig, a, b);
    }
}
/**
 * @generated from message sdk.v1.AgentDefinition
 */
export class AgentDefinition extends Message {
    /**
     * @generated from field: string description = 1;
     */
    description = "";
    /**
     * @generated from field: string prompt = 2;
     */
    prompt = "";
    /**
     * @generated from field: sdk.v1.ModelSelection model = 3;
     */
    model;
    /**
     * @generated from field: optional bool inherit_model = 4;
     */
    inheritModel;
    /**
     * @generated from field: repeated sdk.v1.AgentDefinitionMcpServer mcp_servers = 5;
     */
    mcpServers = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.AgentDefinition";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "description", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "prompt", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "model", kind: "message", T: ModelSelection },
        { no: 4, name: "inherit_model", kind: "scalar", T: 8 /* ScalarType.BOOL */, opt: true },
        { no: 5, name: "mcp_servers", kind: "message", T: AgentDefinitionMcpServer, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new AgentDefinition().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AgentDefinition().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AgentDefinition().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AgentDefinition, a, b);
    }
}
/**
 * @generated from message sdk.v1.AgentDefinitionMcpServer
 */
export class AgentDefinitionMcpServer extends Message {
    /**
     * @generated from oneof sdk.v1.AgentDefinitionMcpServer.value
     */
    value = { case: undefined };
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.AgentDefinitionMcpServer";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */, oneof: "value" },
        { no: 2, name: "inline_config", kind: "message", T: McpServerConfig, oneof: "value" },
    ]);
    static fromBinary(bytes, options) {
        return new AgentDefinitionMcpServer().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AgentDefinitionMcpServer().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AgentDefinitionMcpServer().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AgentDefinitionMcpServer, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkAgentInfo
 */
export class SdkAgentInfo extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: string name = 2;
     */
    name = "";
    /**
     * @generated from field: string summary = 3;
     */
    summary = "";
    /**
     * @generated from field: google.protobuf.Timestamp last_modified = 4;
     */
    lastModified;
    /**
     * @generated from field: sdk.v1.AgentInfoStatus status = 5;
     */
    status = AgentInfoStatus.UNSPECIFIED;
    /**
     * @generated from field: google.protobuf.Timestamp created_at = 6;
     */
    createdAt;
    /**
     * @generated from field: bool archived = 7;
     */
    archived = false;
    /**
     * @generated from oneof sdk.v1.SdkAgentInfo.runtime_info
     */
    runtimeInfo = { case: undefined };
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkAgentInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "summary", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "last_modified", kind: "message", T: Timestamp },
        { no: 5, name: "status", kind: "enum", T: proto3.getEnumType(AgentInfoStatus) },
        { no: 6, name: "created_at", kind: "message", T: Timestamp },
        { no: 7, name: "archived", kind: "scalar", T: 8 /* ScalarType.BOOL */ },
        { no: 8, name: "local", kind: "message", T: LocalAgentInfo, oneof: "runtime_info" },
        { no: 9, name: "cloud", kind: "message", T: CloudAgentInfo, oneof: "runtime_info" },
    ]);
    static fromBinary(bytes, options) {
        return new SdkAgentInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkAgentInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkAgentInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkAgentInfo, a, b);
    }
}
/**
 * @generated from message sdk.v1.LocalAgentInfo
 */
export class LocalAgentInfo extends Message {
    /**
     * @generated from field: string cwd = 1;
     */
    cwd = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.LocalAgentInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "cwd", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new LocalAgentInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new LocalAgentInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new LocalAgentInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(LocalAgentInfo, a, b);
    }
}
/**
 * @generated from message sdk.v1.CloudAgentInfo
 */
export class CloudAgentInfo extends Message {
    /**
     * @generated from field: sdk.v1.CloudEnvironment env = 1;
     */
    env;
    /**
     * @generated from field: repeated string repos = 2;
     */
    repos = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CloudAgentInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "env", kind: "message", T: CloudEnvironment },
        { no: 2, name: "repos", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new CloudAgentInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CloudAgentInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CloudAgentInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CloudAgentInfo, a, b);
    }
}
/**
 * Per-turn or cumulative token counts. total_tokens excludes reasoning_tokens
 * (a visibility-only subset of output), matching the TS SDK TokenUsage shape.
 *
 * @generated from message sdk.v1.TokenUsage
 */
export class TokenUsage extends Message {
    /**
     * @generated from field: int64 input_tokens = 1;
     */
    inputTokens = protoInt64.zero;
    /**
     * @generated from field: int64 output_tokens = 2;
     */
    outputTokens = protoInt64.zero;
    /**
     * @generated from field: int64 cache_read_tokens = 3;
     */
    cacheReadTokens = protoInt64.zero;
    /**
     * @generated from field: int64 cache_write_tokens = 4;
     */
    cacheWriteTokens = protoInt64.zero;
    /**
     * @generated from field: int64 total_tokens = 5;
     */
    totalTokens = protoInt64.zero;
    /**
     * @generated from field: optional int64 reasoning_tokens = 6;
     */
    reasoningTokens;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.TokenUsage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "input_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */ },
        { no: 2, name: "output_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */ },
        { no: 3, name: "cache_read_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */ },
        { no: 4, name: "cache_write_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */ },
        { no: 5, name: "total_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */ },
        { no: 6, name: "reasoning_tokens", kind: "scalar", T: 3 /* ScalarType.INT64 */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new TokenUsage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new TokenUsage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new TokenUsage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(TokenUsage, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunSnapshot
 */
export class RunSnapshot extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    /**
     * @generated from field: string agent_id = 2;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.RunLifecycleStatus status = 3;
     */
    status = RunLifecycleStatus.UNSPECIFIED;
    /**
     * @generated from field: string result = 4;
     */
    result = "";
    /**
     * @generated from field: sdk.v1.ModelSelection model = 5;
     */
    model;
    /**
     * @generated from field: uint64 duration_ms = 6;
     */
    durationMs = protoInt64.zero;
    /**
     * @generated from field: sdk.v1.RunGitInfo git = 7;
     */
    git;
    /**
     * @generated from field: google.protobuf.Timestamp created_at = 8;
     */
    createdAt;
    /**
     * @generated from field: optional sdk.v1.TokenUsage usage = 9;
     */
    usage;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunSnapshot";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "status", kind: "enum", T: proto3.getEnumType(RunLifecycleStatus) },
        { no: 4, name: "result", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "model", kind: "message", T: ModelSelection },
        { no: 6, name: "duration_ms", kind: "scalar", T: 4 /* ScalarType.UINT64 */ },
        { no: 7, name: "git", kind: "message", T: RunGitInfo },
        { no: 8, name: "created_at", kind: "message", T: Timestamp },
        { no: 9, name: "usage", kind: "message", T: TokenUsage, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new RunSnapshot().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunSnapshot().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunSnapshot().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunSnapshot, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunResult
 */
export class RunResult extends Message {
    /**
     * @generated from field: string run_id = 1;
     */
    runId = "";
    /**
     * @generated from field: string agent_id = 2;
     */
    agentId = "";
    /**
     * @generated from field: sdk.v1.RunLifecycleStatus status = 3;
     */
    status = RunLifecycleStatus.UNSPECIFIED;
    /**
     * @generated from field: string result = 4;
     */
    result = "";
    /**
     * @generated from field: sdk.v1.ModelSelection model = 5;
     */
    model;
    /**
     * @generated from field: uint64 duration_ms = 6;
     */
    durationMs = protoInt64.zero;
    /**
     * @generated from field: sdk.v1.RunGitInfo git = 7;
     */
    git;
    /**
     * @generated from field: google.protobuf.Timestamp created_at = 8;
     */
    createdAt;
    /**
     * @generated from field: optional sdk.v1.TokenUsage usage = 9;
     */
    usage;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunResult";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "status", kind: "enum", T: proto3.getEnumType(RunLifecycleStatus) },
        { no: 4, name: "result", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 5, name: "model", kind: "message", T: ModelSelection },
        { no: 6, name: "duration_ms", kind: "scalar", T: 4 /* ScalarType.UINT64 */ },
        { no: 7, name: "git", kind: "message", T: RunGitInfo },
        { no: 8, name: "created_at", kind: "message", T: Timestamp },
        { no: 9, name: "usage", kind: "message", T: TokenUsage, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new RunResult().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunResult().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunResult().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunResult, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunGitInfo
 */
export class RunGitInfo extends Message {
    /**
     * @generated from field: repeated sdk.v1.RunGitBranchInfo branches = 1;
     */
    branches = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunGitInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "branches", kind: "message", T: RunGitBranchInfo, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new RunGitInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunGitInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunGitInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunGitInfo, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunGitBranchInfo
 */
export class RunGitBranchInfo extends Message {
    /**
     * @generated from field: string repo_url = 1;
     */
    repoUrl = "";
    /**
     * @generated from field: string branch = 2;
     */
    branch = "";
    /**
     * @generated from field: string pr_url = 3;
     */
    prUrl = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunGitBranchInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "repo_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "branch", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "pr_url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new RunGitBranchInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunGitBranchInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunGitBranchInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunGitBranchInfo, a, b);
    }
}
/**
 * Stream message envelope for live Send and durable ObserveRun streams.
 *
 * @generated from message sdk.v1.RunStreamMessage
 */
export class RunStreamMessage extends Message {
    /**
     * @generated from oneof sdk.v1.RunStreamMessage.envelope
     */
    envelope = { case: undefined };
    /**
     * Opaque exclusive resume token, scoped to a single run.
     *
     * @generated from field: optional string offset = 4;
     */
    offset;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunStreamMessage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "sdk_message", kind: "message", T: SdkMessage, oneof: "envelope" },
        { no: 2, name: "result", kind: "message", T: RunStreamResult, oneof: "envelope" },
        { no: 3, name: "done", kind: "message", T: RunStreamDone, oneof: "envelope" },
        { no: 5, name: "interaction_update", kind: "message", T: InteractionUpdate, oneof: "envelope" },
        { no: 6, name: "step", kind: "message", T: ConversationStep, oneof: "envelope" },
        { no: 4, name: "offset", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new RunStreamMessage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunStreamMessage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunStreamMessage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunStreamMessage, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunStreamResult
 */
export class RunStreamResult extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: string run_id = 2;
     */
    runId = "";
    /**
     * @generated from field: sdk.v1.RunLifecycleStatus status = 3;
     */
    status = RunLifecycleStatus.UNSPECIFIED;
    /**
     * @generated from field: optional string error_code = 4;
     */
    errorCode;
    /**
     * @generated from field: sdk.v1.RunResult result = 5;
     */
    result;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunStreamResult";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "status", kind: "enum", T: proto3.getEnumType(RunLifecycleStatus) },
        { no: 4, name: "error_code", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 5, name: "result", kind: "message", T: RunResult },
    ]);
    static fromBinary(bytes, options) {
        return new RunStreamResult().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunStreamResult().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunStreamResult().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunStreamResult, a, b);
    }
}
/**
 * @generated from message sdk.v1.RunStreamDone
 */
export class RunStreamDone extends Message {
    /**
     * @generated from field: string agent_id = 1;
     */
    agentId = "";
    /**
     * @generated from field: string run_id = 2;
     */
    runId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RunStreamDone";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "run_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new RunStreamDone().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RunStreamDone().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RunStreamDone().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RunStreamDone, a, b);
    }
}
/**
 * @generated from message sdk.v1.InteractionUpdate
 */
export class InteractionUpdate extends Message {
    /**
     * @generated from field: string type = 1;
     */
    type = "";
    /**
     * @generated from field: google.protobuf.Struct update = 2;
     */
    update;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.InteractionUpdate";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "update", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new InteractionUpdate().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new InteractionUpdate().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new InteractionUpdate().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(InteractionUpdate, a, b);
    }
}
/**
 * @generated from message sdk.v1.ConversationStep
 */
export class ConversationStep extends Message {
    /**
     * @generated from field: string type = 1;
     */
    type = "";
    /**
     * @generated from field: google.protobuf.Struct step = 2;
     */
    step;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ConversationStep";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "step", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new ConversationStep().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ConversationStep().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ConversationStep().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ConversationStep, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkMessage
 */
export class SdkMessage extends Message {
    /**
     * @generated from field: string type = 1;
     */
    type = "";
    /**
     * @generated from field: google.protobuf.Struct message = 2;
     */
    message;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkMessage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "message", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new SdkMessage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkMessage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkMessage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkMessage, a, b);
    }
}
/**
 * @generated from message sdk.v1.AgentMessage
 */
export class AgentMessage extends Message {
    /**
     * @generated from field: string type = 1;
     */
    type = "";
    /**
     * @generated from field: string uuid = 2;
     */
    uuid = "";
    /**
     * @generated from field: string agent_id = 3;
     */
    agentId = "";
    /**
     * @generated from field: google.protobuf.Struct message = 4;
     */
    message;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.AgentMessage";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "uuid", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "message", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new AgentMessage().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new AgentMessage().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new AgentMessage().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(AgentMessage, a, b);
    }
}
/**
 * @generated from message sdk.v1.SdkArtifact
 */
export class SdkArtifact extends Message {
    /**
     * @generated from field: string path = 1;
     */
    path = "";
    /**
     * @generated from field: uint64 size_bytes = 2;
     */
    sizeBytes = protoInt64.zero;
    /**
     * @generated from field: string updated_at = 3;
     */
    updatedAt = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkArtifact";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "path", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "size_bytes", kind: "scalar", T: 4 /* ScalarType.UINT64 */ },
        { no: 3, name: "updated_at", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SdkArtifact().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkArtifact().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkArtifact().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkArtifact, a, b);
    }
}
/**
 * @generated from message sdk.v1.DownloadArtifactChunk
 */
export class DownloadArtifactChunk extends Message {
    /**
     * @generated from field: bytes data = 1;
     */
    data = new Uint8Array(0);
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.DownloadArtifactChunk";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "data", kind: "scalar", T: 12 /* ScalarType.BYTES */ },
    ]);
    static fromBinary(bytes, options) {
        return new DownloadArtifactChunk().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new DownloadArtifactChunk().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new DownloadArtifactChunk().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(DownloadArtifactChunk, a, b);
    }
}
