// @anysphere/proto/dist/generated/sdk/v1/sdk_errors_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_errors.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Duration, Message, proto3 } from "@bufbuild/protobuf";
export { SdkErrorCode };
/**
 * Public SDK error codes returned by the bridge.
 *
 * @generated from enum sdk.v1.SdkErrorCode
 */
var SdkErrorCode;
(function (SdkErrorCode) {
    /**
     * @generated from enum value: SDK_ERROR_CODE_UNSPECIFIED = 0;
     */
    SdkErrorCode[SdkErrorCode["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_UNAUTHORIZED = 1;
     */
    SdkErrorCode[SdkErrorCode["UNAUTHORIZED"] = 1] = "UNAUTHORIZED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_API_KEY_NOT_FOUND = 2;
     */
    SdkErrorCode[SdkErrorCode["API_KEY_NOT_FOUND"] = 2] = "API_KEY_NOT_FOUND";
    /**
     * @generated from enum value: SDK_ERROR_CODE_PLAN_REQUIRED = 3;
     */
    SdkErrorCode[SdkErrorCode["PLAN_REQUIRED"] = 3] = "PLAN_REQUIRED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_ROLE_FORBIDDEN = 4;
     */
    SdkErrorCode[SdkErrorCode["ROLE_FORBIDDEN"] = 4] = "ROLE_FORBIDDEN";
    /**
     * @generated from enum value: SDK_ERROR_CODE_FEATURE_UNAVAILABLE = 5;
     */
    SdkErrorCode[SdkErrorCode["FEATURE_UNAVAILABLE"] = 5] = "FEATURE_UNAVAILABLE";
    /**
     * @generated from enum value: SDK_ERROR_CODE_AGENT_NOT_FOUND = 6;
     */
    SdkErrorCode[SdkErrorCode["AGENT_NOT_FOUND"] = 6] = "AGENT_NOT_FOUND";
    /**
     * @generated from enum value: SDK_ERROR_CODE_RUN_NOT_FOUND = 7;
     */
    SdkErrorCode[SdkErrorCode["RUN_NOT_FOUND"] = 7] = "RUN_NOT_FOUND";
    /**
     * @generated from enum value: SDK_ERROR_CODE_VALIDATION_ERROR = 8;
     */
    SdkErrorCode[SdkErrorCode["VALIDATION_ERROR"] = 8] = "VALIDATION_ERROR";
    /**
     * @generated from enum value: SDK_ERROR_CODE_INVALID_MODEL = 9;
     */
    SdkErrorCode[SdkErrorCode["INVALID_MODEL"] = 9] = "INVALID_MODEL";
    /**
     * @generated from enum value: SDK_ERROR_CODE_INVALID_BRANCH_NAME = 10;
     */
    SdkErrorCode[SdkErrorCode["INVALID_BRANCH_NAME"] = 10] = "INVALID_BRANCH_NAME";
    /**
     * @generated from enum value: SDK_ERROR_CODE_REPOSITORY_REQUIRED = 11;
     */
    SdkErrorCode[SdkErrorCode["REPOSITORY_REQUIRED"] = 11] = "REPOSITORY_REQUIRED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_REPOSITORY_ACCESS = 12;
     */
    SdkErrorCode[SdkErrorCode["REPOSITORY_ACCESS"] = 12] = "REPOSITORY_ACCESS";
    /**
     * @generated from enum value: SDK_ERROR_CODE_PR_RESOLUTION_FAILED = 13;
     */
    SdkErrorCode[SdkErrorCode["PR_RESOLUTION_FAILED"] = 13] = "PR_RESOLUTION_FAILED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_USAGE_LIMIT_EXCEEDED = 14;
     */
    SdkErrorCode[SdkErrorCode["USAGE_LIMIT_EXCEEDED"] = 14] = "USAGE_LIMIT_EXCEEDED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_AGENT_BUSY = 15;
     */
    SdkErrorCode[SdkErrorCode["AGENT_BUSY"] = 15] = "AGENT_BUSY";
    /**
     * @generated from enum value: SDK_ERROR_CODE_AGENT_ARCHIVED = 16;
     */
    SdkErrorCode[SdkErrorCode["AGENT_ARCHIVED"] = 16] = "AGENT_ARCHIVED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_RUN_NOT_CANCELLABLE = 17;
     */
    SdkErrorCode[SdkErrorCode["RUN_NOT_CANCELLABLE"] = 17] = "RUN_NOT_CANCELLABLE";
    /**
     * @generated from enum value: SDK_ERROR_CODE_RATE_LIMIT_EXCEEDED = 18;
     */
    SdkErrorCode[SdkErrorCode["RATE_LIMIT_EXCEEDED"] = 18] = "RATE_LIMIT_EXCEEDED";
    /**
     * @generated from enum value: SDK_ERROR_CODE_UPSTREAM_ERROR = 19;
     */
    SdkErrorCode[SdkErrorCode["UPSTREAM_ERROR"] = 19] = "UPSTREAM_ERROR";
    /**
     * @generated from enum value: SDK_ERROR_CODE_INTERNAL_ERROR = 20;
     */
    SdkErrorCode[SdkErrorCode["INTERNAL_ERROR"] = 20] = "INTERNAL_ERROR";
    /**
     * @generated from enum value: SDK_ERROR_CODE_CLIENT_CANCELLED = 21;
     */
    SdkErrorCode[SdkErrorCode["CLIENT_CANCELLED"] = 21] = "CLIENT_CANCELLED";
})(SdkErrorCode || (SdkErrorCode = {}));
// Retrieve enum metadata with: proto3.getEnumType(SdkErrorCode)
proto3.util.setEnumType(SdkErrorCode, "sdk.v1.SdkErrorCode", [
    { no: 0, name: "SDK_ERROR_CODE_UNSPECIFIED" },
    { no: 1, name: "SDK_ERROR_CODE_UNAUTHORIZED" },
    { no: 2, name: "SDK_ERROR_CODE_API_KEY_NOT_FOUND" },
    { no: 3, name: "SDK_ERROR_CODE_PLAN_REQUIRED" },
    { no: 4, name: "SDK_ERROR_CODE_ROLE_FORBIDDEN" },
    { no: 5, name: "SDK_ERROR_CODE_FEATURE_UNAVAILABLE" },
    { no: 6, name: "SDK_ERROR_CODE_AGENT_NOT_FOUND" },
    { no: 7, name: "SDK_ERROR_CODE_RUN_NOT_FOUND" },
    { no: 8, name: "SDK_ERROR_CODE_VALIDATION_ERROR" },
    { no: 9, name: "SDK_ERROR_CODE_INVALID_MODEL" },
    { no: 10, name: "SDK_ERROR_CODE_INVALID_BRANCH_NAME" },
    { no: 11, name: "SDK_ERROR_CODE_REPOSITORY_REQUIRED" },
    { no: 12, name: "SDK_ERROR_CODE_REPOSITORY_ACCESS" },
    { no: 13, name: "SDK_ERROR_CODE_PR_RESOLUTION_FAILED" },
    { no: 14, name: "SDK_ERROR_CODE_USAGE_LIMIT_EXCEEDED" },
    { no: 15, name: "SDK_ERROR_CODE_AGENT_BUSY" },
    { no: 16, name: "SDK_ERROR_CODE_AGENT_ARCHIVED" },
    { no: 17, name: "SDK_ERROR_CODE_RUN_NOT_CANCELLABLE" },
    { no: 18, name: "SDK_ERROR_CODE_RATE_LIMIT_EXCEEDED" },
    { no: 19, name: "SDK_ERROR_CODE_UPSTREAM_ERROR" },
    { no: 20, name: "SDK_ERROR_CODE_INTERNAL_ERROR" },
    { no: 21, name: "SDK_ERROR_CODE_CLIENT_CANCELLED" },
]);
/**
 * Rate-limit metadata surfaced from Cursor Cloud responses.
 *
 * @generated from message sdk.v1.RateLimitInfo
 */
export class RateLimitInfo extends Message {
    /**
     * @generated from field: optional uint64 limit = 1;
     */
    limit;
    /**
     * @generated from field: optional uint64 remaining = 2;
     */
    remaining;
    /**
     * @generated from field: optional uint64 reset_epoch_seconds = 3;
     */
    resetEpochSeconds;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.RateLimitInfo";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "limit", kind: "scalar", T: 4 /* ScalarType.UINT64 */, opt: true },
        { no: 2, name: "remaining", kind: "scalar", T: 4 /* ScalarType.UINT64 */, opt: true },
        { no: 3, name: "reset_epoch_seconds", kind: "scalar", T: 4 /* ScalarType.UINT64 */, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new RateLimitInfo().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new RateLimitInfo().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new RateLimitInfo().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(RateLimitInfo, a, b);
    }
}
/**
 * Stable, public error details attached to Connect errors.
 *
 * @generated from message sdk.v1.SdkErrorDetails
 */
export class SdkErrorDetails extends Message {
    /**
     * Full request ID from Cursor Cloud, if one is available. Never truncate this.
     *
     * @generated from field: optional string request_id = 1;
     */
    requestId;
    /**
     * @generated from field: sdk.v1.SdkErrorCode sdk_error_code = 2;
     */
    sdkErrorCode = SdkErrorCode.UNSPECIFIED;
    /**
     * @generated from field: string message = 3;
     */
    message = "";
    /**
     * @generated from field: optional string help_url = 4;
     */
    helpUrl;
    /**
     * @generated from field: optional string provider = 5;
     */
    provider;
    /**
     * @generated from field: optional google.protobuf.Duration retry_after = 6;
     */
    retryAfter;
    /**
     * @generated from field: optional sdk.v1.RateLimitInfo rate_limit = 7;
     */
    rateLimit;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SdkErrorDetails";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "request_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 2, name: "sdk_error_code", kind: "enum", T: proto3.getEnumType(SdkErrorCode) },
        { no: 3, name: "message", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 4, name: "help_url", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 5, name: "provider", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 6, name: "retry_after", kind: "message", T: Duration, opt: true },
        { no: 7, name: "rate_limit", kind: "message", T: RateLimitInfo, opt: true },
    ]);
    static fromBinary(bytes, options) {
        return new SdkErrorDetails().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SdkErrorDetails().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SdkErrorDetails().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SdkErrorDetails, a, b);
    }
}
