// @anysphere/proto/dist/generated/sdk/v1/sdk_custom_tool_callback_service_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_custom_tool_callback_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3, Struct } from "@bufbuild/protobuf";
/**
 * @generated from message sdk.v1.CallCustomToolRequest
 */
export class CallCustomToolRequest extends Message {
    /**
     * @generated from field: string tool_name = 1;
     */
    toolName = "";
    /**
     * @generated from field: google.protobuf.Struct args = 2;
     */
    args;
    /**
     * @generated from field: optional string tool_call_id = 3;
     */
    toolCallId;
    /**
     * Agent that owns the tool definition (CreateAgent / ResumeAgent local.custom_tools).
     *
     * @generated from field: string agent_id = 4;
     */
    agentId = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CallCustomToolRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "tool_name", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "args", kind: "message", T: Struct },
        { no: 3, name: "tool_call_id", kind: "scalar", T: 9 /* ScalarType.STRING */, opt: true },
        { no: 4, name: "agent_id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new CallCustomToolRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CallCustomToolRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CallCustomToolRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CallCustomToolRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CallCustomToolResponse
 */
export class CallCustomToolResponse extends Message {
    /**
     * SDKCustomToolResult wire shape (string, object, or content envelope).
     *
     * @generated from field: google.protobuf.Struct result = 1;
     */
    result;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CallCustomToolResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "result", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new CallCustomToolResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CallCustomToolResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CallCustomToolResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CallCustomToolResponse, a, b);
    }
}
