// @anysphere/proto/dist/generated/sdk/v1/sdk_cursor_service_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_cursor_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3 } from "@bufbuild/protobuf";
import { SdkModel, SdkRepository, SdkUser } from "./sdk_messages_pb.js";
/**
 * @generated from message sdk.v1.CursorRequestOptions
 */
export class CursorRequestOptions extends Message {
    /**
     * @generated from field: string api_key = 1;
     */
    apiKey = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CursorRequestOptions";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "api_key", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new CursorRequestOptions().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CursorRequestOptions().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CursorRequestOptions().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CursorRequestOptions, a, b);
    }
}
/**
 * @generated from message sdk.v1.MeRequest
 */
export class MeRequest extends Message {
    /**
     * @generated from field: sdk.v1.CursorRequestOptions options = 1;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.MeRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "options", kind: "message", T: CursorRequestOptions },
    ]);
    static fromBinary(bytes, options) {
        return new MeRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MeRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MeRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(MeRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.MeResponse
 */
export class MeResponse extends Message {
    /**
     * @generated from field: sdk.v1.SdkUser user = 1;
     */
    user;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.MeResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "user", kind: "message", T: SdkUser },
    ]);
    static fromBinary(bytes, options) {
        return new MeResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new MeResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new MeResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(MeResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListModelsRequest
 */
export class ListModelsRequest extends Message {
    /**
     * @generated from field: sdk.v1.CursorRequestOptions options = 1;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListModelsRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "options", kind: "message", T: CursorRequestOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ListModelsRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListModelsRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListModelsRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListModelsRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListModelsResponse
 */
export class ListModelsResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.SdkModel items = 1;
     */
    items = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListModelsResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "items", kind: "message", T: SdkModel, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ListModelsResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListModelsResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListModelsResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListModelsResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListRepositoriesRequest
 */
export class ListRepositoriesRequest extends Message {
    /**
     * @generated from field: sdk.v1.CursorRequestOptions options = 1;
     */
    options;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListRepositoriesRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "options", kind: "message", T: CursorRequestOptions },
    ]);
    static fromBinary(bytes, options) {
        return new ListRepositoriesRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRepositoriesRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRepositoriesRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListRepositoriesRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ListRepositoriesResponse
 */
export class ListRepositoriesResponse extends Message {
    /**
     * @generated from field: repeated sdk.v1.SdkRepository items = 1;
     */
    items = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ListRepositoriesResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "items", kind: "message", T: SdkRepository, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new ListRepositoriesResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ListRepositoriesResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ListRepositoriesResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ListRepositoriesResponse, a, b);
    }
}
