// @anysphere/proto/dist/generated/sdk/v1/sdk_store_callback_service_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_store_callback_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3, Struct } from "@bufbuild/protobuf";
/**
 * Generic LocalAgentStore operation forwarded to a host-owned substore.
 *
 * @generated from message sdk.v1.CallStoreRequest
 */
export class CallStoreRequest extends Message {
    /**
     * Substore being addressed: "agents", "runs", "runEvents", or "checkpoints".
     *
     * @generated from field: string substore = 1;
     */
    substore = "";
    /**
     * Operation on the substore: "get", "create", "update", "delete", "list", or
     * "append" (runEvents only).
     *
     * @generated from field: string method = 2;
     */
    method = "";
    /**
     * Operation input object. Checkpoint blob bytes are base64 strings.
     *
     * @generated from field: google.protobuf.Struct input = 3;
     */
    input;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CallStoreRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "substore", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "method", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "input", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new CallStoreRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CallStoreRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CallStoreRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CallStoreRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.CallStoreResponse
 */
export class CallStoreResponse extends Message {
    /**
     * Operation output object, or unset for a null result (e.g. a `get` miss or
     * a `delete`). Checkpoint blob bytes are base64 strings.
     *
     * @generated from field: google.protobuf.Struct output = 1;
     */
    output;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.CallStoreResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "output", kind: "message", T: Struct },
    ]);
    static fromBinary(bytes, options) {
        return new CallStoreResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new CallStoreResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new CallStoreResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(CallStoreResponse, a, b);
    }
}
