// @anysphere/proto/dist/generated/sdk/v1/sdk_store_callback_service_connect.js
// @generated by protoc-gen-connect-es v1.6.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_store_callback_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { CallStoreRequest, CallStoreResponse } from "./sdk_store_callback_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
/**
 * Store callbacks from the Node bridge to a Python (or other) host process.
 *
 * The bridge forwards local-agent-store operations to the host over a single
 * generic `CallStore` RPC, mirroring the `@cursor/sdk` `LocalAgentStore`
 * topology (substores: agents, runs, runEvents, checkpoints). Inputs/outputs
 * are structured JSON so the bridge can forward the full surface without a
 * typed message per operation. This powers `local.store` type "custom", where
 * every substore is implemented by the host process.
 *
 * @generated from service sdk.v1.SdkStoreCallbackService
 */
export const SdkStoreCallbackService = {
    typeName: "sdk.v1.SdkStoreCallbackService",
    methods: {
        /**
         * @generated from rpc sdk.v1.SdkStoreCallbackService.CallStore
         */
        callStore: {
            name: "CallStore",
            I: CallStoreRequest,
            O: CallStoreResponse,
            kind: MethodKind.Unary,
        },
    }
};
