// @anysphere/proto/dist/generated/sdk/v1/sdk_bridge_control_service_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_bridge_control_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { Message, proto3 } from "@bufbuild/protobuf";
/**
 * @generated from message sdk.v1.PingRequest
 */
export class PingRequest extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.PingRequest";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new PingRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PingRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PingRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(PingRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.PingResponse
 */
export class PingResponse extends Message {
    /**
     * @generated from field: string message = 1;
     */
    message = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.PingResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "message", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new PingResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new PingResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new PingResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(PingResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.ShutdownRequest
 */
export class ShutdownRequest extends Message {
    /**
     * @generated from field: uint32 grace_seconds = 1;
     */
    graceSeconds = 0;
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ShutdownRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "grace_seconds", kind: "scalar", T: 13 /* ScalarType.UINT32 */ },
    ]);
    static fromBinary(bytes, options) {
        return new ShutdownRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ShutdownRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ShutdownRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ShutdownRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.ShutdownResponse
 */
export class ShutdownResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.ShutdownResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new ShutdownResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new ShutdownResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new ShutdownResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(ShutdownResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetVersionRequest
 */
export class GetVersionRequest extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetVersionRequest";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new GetVersionRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetVersionRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetVersionRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetVersionRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.GetVersionResponse
 */
export class GetVersionResponse extends Message {
    /**
     * @generated from field: string bridge_version = 1;
     */
    bridgeVersion = "";
    /**
     * @generated from field: string protocol_version = 2;
     */
    protocolVersion = "";
    /**
     * @generated from field: repeated string capabilities = 3;
     */
    capabilities = [];
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.GetVersionResponse";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "bridge_version", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "protocol_version", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 3, name: "capabilities", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    ]);
    static fromBinary(bytes, options) {
        return new GetVersionResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new GetVersionResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new GetVersionResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(GetVersionResponse, a, b);
    }
}
/**
 * @generated from message sdk.v1.SetToolCallbackRequest
 */
export class SetToolCallbackRequest extends Message {
    /**
     * Connect base URL of the host custom-tool callback server. Empty clears the
     * currently configured callback.
     *
     * @generated from field: string url = 1;
     */
    url = "";
    /**
     * Bearer token the bridge presents on CallCustomTool RPCs.
     *
     * @generated from field: string auth_token = 2;
     */
    authToken = "";
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SetToolCallbackRequest";
    static fields = proto3.util.newFieldList(() => [
        { no: 1, name: "url", kind: "scalar", T: 9 /* ScalarType.STRING */ },
        { no: 2, name: "auth_token", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    ]);
    static fromBinary(bytes, options) {
        return new SetToolCallbackRequest().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SetToolCallbackRequest().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SetToolCallbackRequest().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SetToolCallbackRequest, a, b);
    }
}
/**
 * @generated from message sdk.v1.SetToolCallbackResponse
 */
export class SetToolCallbackResponse extends Message {
    constructor(data) {
        super();
        proto3.util.initPartial(data, this);
    }
    static runtime = proto3;
    static typeName = "sdk.v1.SetToolCallbackResponse";
    static fields = proto3.util.newFieldList(() => []);
    static fromBinary(bytes, options) {
        return new SetToolCallbackResponse().fromBinary(bytes, options);
    }
    static fromJson(jsonValue, options) {
        return new SetToolCallbackResponse().fromJson(jsonValue, options);
    }
    static fromJsonString(jsonString, options) {
        return new SetToolCallbackResponse().fromJsonString(jsonString, options);
    }
    static equals(a, b) {
        return proto3.util.equals(SetToolCallbackResponse, a, b);
    }
}
