// @anysphere/proto/dist/generated/sdk/v1/sdk_bridge_control_service_connect.js
// @generated by protoc-gen-connect-es v1.6.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_bridge_control_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { GetVersionRequest, GetVersionResponse, PingRequest, PingResponse, SetToolCallbackRequest, SetToolCallbackResponse, ShutdownRequest, ShutdownResponse } from "./sdk_bridge_control_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
/**
 * Bridge-local health, version, and lifecycle APIs.
 *
 * @generated from service sdk.v1.SdkBridgeControlService
 */
export const SdkBridgeControlService = {
    typeName: "sdk.v1.SdkBridgeControlService",
    methods: {
        /**
         * @generated from rpc sdk.v1.SdkBridgeControlService.Ping
         */
        ping: {
            name: "Ping",
            I: PingRequest,
            O: PingResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkBridgeControlService.Shutdown
         */
        shutdown: {
            name: "Shutdown",
            I: ShutdownRequest,
            O: ShutdownResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkBridgeControlService.GetVersion
         */
        getVersion: {
            name: "GetVersion",
            I: GetVersionRequest,
            O: GetVersionResponse,
            kind: MethodKind.Unary,
        },
        /**
         * Point the bridge at a host-process custom-tool callback server after
         * startup. Lets SDK clients that attach to an already-running bridge
         * (Client.connect) register host execute handlers. Same-host (loopback) only.
         *
         * @generated from rpc sdk.v1.SdkBridgeControlService.SetToolCallback
         */
        setToolCallback: {
            name: "SetToolCallback",
            I: SetToolCallbackRequest,
            O: SetToolCallbackResponse,
            kind: MethodKind.Unary,
        },
    }
};
