// @anysphere/proto/dist/generated/sdk/v1/sdk_agent_service_connect.js
// @generated by protoc-gen-connect-es v1.6.1 with parameter "target=ts"
// @generated from file sdk/v1/sdk_agent_service.proto (package sdk.v1, syntax proto3)
/* eslint-disable */
// @ts-nocheck
import { ArchiveAgentRequest, ArchiveAgentResponse, CancelRunRequest, CancelRunResponse, CloseAgentRequest, CloseAgentResponse, CreateAgentRequest, CreateAgentResponse, DeleteAgentRequest, DeleteAgentResponse, DownloadArtifactRequest, GetAgentRequest, GetAgentResponse, GetRunConversationRequest, GetRunConversationResponse, GetRunRequest, GetRunResponse, ListAgentMessagesRequest, ListAgentMessagesResponse, ListAgentsRequest, ListAgentsResponse, ListArtifactsRequest, ListArtifactsResponse, ListRunsRequest, ListRunsResponse, ObserveRunRequest, ReloadAgentRequest, ReloadAgentResponse, ResumeAgentRequest, ResumeAgentResponse, SendRequest, UnarchiveAgentRequest, UnarchiveAgentResponse, WaitLiveRunRequest, WaitLiveRunResponse } from "./sdk_agent_service_pb.js";
import { MethodKind } from "@bufbuild/protobuf";
import { DownloadArtifactChunk, RunStreamMessage } from "./sdk_messages_pb.js";
/**
 * Agent, run, streaming, cancellation, and artifact APIs.
 *
 * @generated from service sdk.v1.SdkAgentService
 */
export const SdkAgentService = {
    typeName: "sdk.v1.SdkAgentService",
    methods: {
        /**
         * @generated from rpc sdk.v1.SdkAgentService.CreateAgent
         */
        createAgent: {
            name: "CreateAgent",
            I: CreateAgentRequest,
            O: CreateAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ResumeAgent
         */
        resumeAgent: {
            name: "ResumeAgent",
            I: ResumeAgentRequest,
            O: ResumeAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ReloadAgent
         */
        reloadAgent: {
            name: "ReloadAgent",
            I: ReloadAgentRequest,
            O: ReloadAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.CloseAgent
         */
        closeAgent: {
            name: "CloseAgent",
            I: CloseAgentRequest,
            O: CloseAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.Send
         */
        send: {
            name: "Send",
            I: SendRequest,
            O: RunStreamMessage,
            kind: MethodKind.ServerStreaming,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.WaitLiveRun
         */
        waitLiveRun: {
            name: "WaitLiveRun",
            I: WaitLiveRunRequest,
            O: WaitLiveRunResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.GetRun
         */
        getRun: {
            name: "GetRun",
            I: GetRunRequest,
            O: GetRunResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ListRuns
         */
        listRuns: {
            name: "ListRuns",
            I: ListRunsRequest,
            O: ListRunsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.GetRunConversation
         */
        getRunConversation: {
            name: "GetRunConversation",
            I: GetRunConversationRequest,
            O: GetRunConversationResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ObserveRun
         */
        observeRun: {
            name: "ObserveRun",
            I: ObserveRunRequest,
            O: RunStreamMessage,
            kind: MethodKind.ServerStreaming,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.CancelRun
         */
        cancelRun: {
            name: "CancelRun",
            I: CancelRunRequest,
            O: CancelRunResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.GetAgent
         */
        getAgent: {
            name: "GetAgent",
            I: GetAgentRequest,
            O: GetAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ListAgents
         */
        listAgents: {
            name: "ListAgents",
            I: ListAgentsRequest,
            O: ListAgentsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ArchiveAgent
         */
        archiveAgent: {
            name: "ArchiveAgent",
            I: ArchiveAgentRequest,
            O: ArchiveAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.UnarchiveAgent
         */
        unarchiveAgent: {
            name: "UnarchiveAgent",
            I: UnarchiveAgentRequest,
            O: UnarchiveAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.DeleteAgent
         */
        deleteAgent: {
            name: "DeleteAgent",
            I: DeleteAgentRequest,
            O: DeleteAgentResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ListAgentMessages
         */
        listAgentMessages: {
            name: "ListAgentMessages",
            I: ListAgentMessagesRequest,
            O: ListAgentMessagesResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.ListArtifacts
         */
        listArtifacts: {
            name: "ListArtifacts",
            I: ListArtifactsRequest,
            O: ListArtifactsResponse,
            kind: MethodKind.Unary,
        },
        /**
         * @generated from rpc sdk.v1.SdkAgentService.DownloadArtifact
         */
        downloadArtifact: {
            name: "DownloadArtifact",
            I: DownloadArtifactRequest,
            O: DownloadArtifactChunk,
            kind: MethodKind.ServerStreaming,
        },
    }
};
